lib/manager/components/DeploymentVersionsTable.js

```diff
@@ -0,0 +1,159 @@
+import React, {PropTypes, Component} from 'react'
+import {Table, Button, Badge, Glyphicon, ButtonToolbar} from 'react-bootstrap'
+import { Link } from 'react-router'
+
+import { getComponentMessages, getMessage } from '../../common/util/config'
+import {versionHasExpired, versionHasNotBegun} from '../util'
+import {formatTimestamp, fromNow} from '../../common/util/date-time'
+
+export default class DeploymentVersionsTable extends Component {
+  static propTypes = {
+  }
+
+  render () {
+    const messages = getComponentMessages('DeploymentViewer')
+    const {deployment, deleteFeedVersion, updateVersionForFeedSource, user} = this.props
+    const {feedVersions: versions} = deployment
+    return (
+      <Table striped hover fill>
+        <thead>
+          <tr>
+            <th className='col-md-4'>{getMessage(messages, 'table.name')}</th>
+            <th>Version</th>
+            <th className='hidden-xs'>{getMessage(messages, 'table.dateRetrieved')}</th>
+            <th className='hidden-xs'>{getMessage(messages, 'table.loadStatus')}</th>
+            <th className='hidden-xs'>{getMessage(messages, 'table.errorCount')}</th>
+            <th className='hidden-xs'>{getMessage(messages, 'table.routeCount')}</th>
+            <th className='hidden-xs'>{getMessage(messages, 'table.tripCount')}</th>
+            <th className='hidden-xs'>{getMessage(messages, 'table.stopTimesCount')}</th>
+            <th>{getMessage(messages, 'table.validFrom')}</th>
+            <th>{getMessage(messages, 'table.expires')}</th>
+            <th />
+          </tr>
+        </thead>
+        <tbody>
+          {versions.map((version) => {
+            return <FeedVersionTableRow
+              feedSource={version.feedSource}
+              version={version}
+              project={deployment.project}
+              deployment={deployment}
+              key={version.id}
+              user={user}
+              updateVersionForFeedSource={updateVersionForFeedSource}
+              deleteFeedVersionClicked={deleteFeedVersion} />
+          })}
+        </tbody>
+      </Table>
+    )
+  }
+}
+
+class FeedVersionTableRow extends Component {
+  static propTypes = {
```

Comment:
We should be heading in the direction of flow typing components.  Perhaps we can begin doing so after this PR.

Reply:
Yea, the next PR we need to address is the Auth0 deprecation.  Then perhaps we can focus on general housekeeping items.